modules/core/src/main/scala/CharCollector.scala

```diff
@@ -51,14 +51,19 @@ private object CharCollector:
     curState match
       case State.Init =>
         char match
+          case 3|4 => // Ctrl+C or Ctrl+D
+            System.exit(1)
+            throw Exception("Got SIGINT or SIGTERM") // should be unreachalbe, but System.exit returns Unit
```

Comment:
May be just return `DecodeResult.Error` here?
System.exit doesn't exist on scala.js it seems: https://github.com/neandertech/cue4s/actions/runs/12338800294/job/34583327970?pr=36#step:5:226

welcome to the awful world of multi-OS multi-platform development 😬

Reply:
Updated in [ce1f252](https://github.com/neandertech/cue4s/pull/36/commits/ce1f2527cd5b5cc0432f7e2708337e6dd3bb131b)

I ended up throwing a `ControlException with NoStackTrace`. I tried the `DecodeResult.Error`, but it was a bit weird (the prompt fails, but the program continues running).

I think it doesn't look that bad without a stacktrace:
```
? How was your day?  ? great
Exception in thread "main" cue4s.CharCollector$ExitThrowable$: Received termination signal
```

Reply:
I'll have a think about this a bit more – realistically the prompt should handle the cancellation and pass that back to the user program – a second cancellation should be sent to the user program to handle/ignore (in latter case it will quit).

Anyways, not a blocker, just needs a bit more design work probably.